ui/src/notebooks/context/notebook.tsx

```diff
@@ -11,22 +14,26 @@ export interface NotebookContextType {
   id: string
   pipes: PipeData[]
   meta: PipeMeta[] // data only used for the view layer for Notebooks
+  listScrollPosition: number
   addPipe: (pipe: PipeData) => void
   updatePipe: (idx: number, pipe: PipeData) => void
   updateMeta: (idx: number, pipe: PipeMeta) => void
   movePipe: (currentIdx: number, newIdx: number) => void
   removePipe: (idx: number) => void
+  scrollToPipe: (idx: number) => void
```

Comment:
I could see grouping these into a single field for readability, but more a matter of opinion

```tsx
scrolling: {
  listScrollPosition: number
  scrollToPipe: (idx: number) => void
},
```

Reply:
instead of a single field, it might make more sense to put them in their own context? it feels like a very different _kind_ of data on the notebook. I got the feeling that pipes and meta should be split into their own contexts when i started adding `results` to it using the same pattern, but they're coupled on array index for now and have to be moved together until we can break them apart